src/lib/sql/sql_translator.cpp

```diff
@@ -441,8 +439,33 @@ std::shared_ptr<AbstractLQPNode> SQLTranslator::_translate_cross_product(const s
   return product;
 }
 
+std::shared_ptr<AbstractLQPNode> SQLTranslator::_apply_column_renames(const std::shared_ptr<AbstractLQPNode> node,
+                                                                      const hsql::TableRef& table) {
+  // Add a new projection node for table alias with column alias declarations
+  // e.g. select * from foo as bar(a,b)
+  if (table.alias && table.alias->columns) {
+    // To stick to the sql standard there must be an alias for every column
+    // of the renamed table
+    // https://www.contrib.andrew.cmu.edu/~shadow/sql/sql1992.txt 6.3
+    Assert(table.alias->columns->size() == node->output_column_count(),
+           "The number of column aliases must match the number of columns");
+    auto& column_references = node->output_column_references();
+    std::vector<std::shared_ptr<LQPExpression>> projections;
+    size_t column_id = 0;
+    for (const char* column : *(table.alias->columns)) {
+      auto alias = std::optional<std::string>(std::string(column));
```

Comment:
No need to create an optional here. You can simply pass the std::string and have it automatically wrapped into an optional.